wallet/src/sync.rs

```diff
@@ -456,3 +467,12 @@ pub(crate) fn get_balances(db: &Db) -> anyhow::Result<impl Iterator<Item = (H256
 
     Ok(balances.into_iter())
 }
+
+pub(crate) fn get_timestamp(db: &Db) -> anyhow::Result<u64> {
+    let timestamp_tree = db.open_tree(TIMESTAMP)?;
+    let timestamp = timestamp_tree
+        .get([0])?
+        .ok_or_else(|| anyhow!("Could not find timestamp in database."))?;
+    u64::decode(&mut &timestamp[..])
+        .map_err(|_| anyhow!("Could not decode timestamp from database."))
+}
```

Comment:
Ideally this would go in a separate timestamp module. My idea is that logic associated with each piece (like how this is associated with the timestamp piece) will live in a dedicated file. That should make it easy to make chain-specific wallets by either including or not including that part of the wallet.